state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -57,6 +59,16 @@ pub struct Swap {
 	pub fee_taken: bool,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct SwapLegInfo {
+	pub swap_id: SwapId,
+	pub from: Asset,
+	pub to: Asset,
+	pub amount: AssetAmount,
+	pub swap_type: SwapType,
```

Comment:
Had a discussion with Kyle and Dan, we agreed that we don't want to expose the SwapType.

Also this means you don't need the serde derives on the SwapType enum.